### ISSUE TITLE
[ban-ts-comment] Enforce `@ts-expect-error: TS1234 because xyz`

### ISSUE THREAD

**Issue by fregante** [typescript-eslint/typescript-eslint#4253]
**Feature request**

I use `ban-ts-comment` with `ts-expect-error` so I must include a description every time, which is great!

```ts
// @ts-expect-error The initial types were incorrect
```

but every time I encounter this it's not very actionable because I don't know what error it's expecting. I think it should always include the exact error ID:

```ts
// @ts-expect-error TS2717 The initial types were incorrect
```

Could this be an option?

```js
{
	'ts-expect-error': 'allow-with-description-and-id',
	minimumDescriptionLength: 10,
}
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.5.0` |
| `TypeScript`                       | `4.5.0` |

**Comment by bradzacher** [comment]
The problem with including error IDs is that  they are not necessarily consistent between TS versions. This is the same reason that the TS team has not added `@ts-expect-error[2717]` or similar. i.e. enforcing that you include an error code could lead to stale/incorrect comments.

I'm not sure if this is a good reason to block this functionality.

**Comment by fregante** [comment]
> The problem with including error IDs is that they are not necessarily consistent between TS versions.

Isn't that information still better than nothing? It's not like an error ID completely changes meaning, I suppose.

**Comment by orta** [comment]
I hit this with twoslash code samples in the TS website, and maybe in the last year for all code samples I've probably changed ~10 error codes. I'd say they're stable _enough_ and maybe it's interesting to note if they do change anyway (though often its just singular to plurals or something)

**Comment by nandi95** [comment]
Links to:
https://github.com/microsoft/TypeScript/issues/45937
https://github.com/microsoft/TypeScript/issues/19139

**Comment by fregante** [comment]
Thanks for the links! It turns out I had already +1'd both at some point in the past 😅 

Anyway here I'm not asking to validate the error code but just to provide it to add a bit more context. Technically the user could enter TS00000 and it would still pass. This is why it's not exactly the same.

**Comment by Josh-Cena** [comment]
IMO error codes are meant for machines, not humans (at least I never got familiar with >3 error codes). If they don't provide any compile-time semantics, do they really help?

**Comment by fregante** [comment]
What do you mean? They're absolutely meant for humans, I can use it to Google an explanation and possible fix

**Comment by Josh-Cena** [comment]
`// @ts-expect-error TS2717` is useful as a TS semantics because it means you can suppress errors more granularly; but not so much for humans. For me personally, `TS2717` gives little more information as I have no idea what error that corresponds to. If I go to the trouble of Googling, while don't I just include the error text in the explanation, or just uncomment that line to see what comes up? Not to dismiss this proposal, just doubting if it should be in the ts-eslint realm

**Comment by bradzacher** [comment]
This is where I much prefer flow's model. Each error flow emits has a human-readable string error code.
For example - flow errors if you attempt to `.bind` a method (because it's unsafe), and the error code for that error is [`method-unbinding`](https://flow.org/try/#0MYGwhgzhAEBiD29oG8BQ0PQEZgE4C5oIAXXASwDsBzaAXmgHIAzRBgbnUwFsBTYgC3gATABQBKFJ0yYBZCADocueSB7UBHaQF9UO0JBgAhPCh2oRFHgHc4icWPm8BwxZVGWbx3PbZA). Then the flow suppression comment is this: `// $FlowExpectedError[method-unbinding]`.

TS goes in a very different direction (which I don't hugely like the ergonomics of), and I would agree with Josh here - `// @ts-expect-error TS2717` would be a pretty cryptic / opaque.
But I also agree that it is nice in terms of "If I delete this comment I'll see a TS2717 error" is a nice property as well.

I'll leave this open in case anyone wants to implement it.

**Comment by JoshuaKGoldberg** [comment]
Relevant discussion: https://github.com/microsoft/TypeScript/issues/45937 & https://github.com/microsoft/TypeScript/issues/19139
